### ISSUE TITLE
Deliberate and explicit usage of error properties

### ISSUE THREAD

**Issue by Xayanide** [Meowhal/osu-ahr#89]
Group: `Refactorization`
- [x] #88
- [x] #92 Deliberate and explicit usage of [Error.prototype.message](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/message) and [Error.prototype.stack](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack)
> For try/catches that **only needs** it. Shouldn't touch anything that aren't supposed to be changed (e.g tests). Should also check out for `error.message` that needs an optional chaining operator (`.?`)

Size: `Medium`
> For end users to show more sorted & informative errors on their issues and also developers to track problems much less difficult than it already is.
> This might be unnecessary.
> e.g 
> ```diff
> -       logger.error('Error: ' + e);
> +       logger.error(`Error:\n${e.message}\n${e.stack}`);
> ```
> Will also deliberately use a newline character (`\n`) after any strings (`Error:`) for error readability.
> ```text
> [ERROR]: Error: Couldn't eat the fruit, apple was rotten... // long error stack
> ```
> ```text
> [ERROR]: Error:
> Couldn't eat the fruit, apple was rotten
> // long error stack
> ```
> This area is quite broad, so, the `try caught` errors that were purposely unhandled, handled etc. will not be touched, in respect to the original intentions.

I will be making a PR for this sometime soon.
If against, please state a concise statement why.

**Comment by Meowhal** [comment]
I am frustrated that javascript can throw objects that are not errors.
I always got tired of thinking about it and did it in a messy way.

If you would be willing to write a PR I would welcome it.

**Comment by Meowhal** [comment]
It may be redundant to include Error in our message since log4js shows ERROR.

```
logger.error(`${e.message}\n${e.stack}`);
```

```
[ERROR]: Couldn't eat the fruit, apple was rotten
// long error stack
```

**Comment by Xayanide** [comment]
> It may be redundant to include Error in our message since log4js shows ERROR.

> ```
In that case, it was indeed redundant, that was an example just in case there was a string like:

```
[ERROR]: @Class#Method
Couldn't eat the fruit, apple was rotten
// long error stack
```

**Comment by Xayanide** [comment]
> It may be redundant to include Error in our message since log4js shows ERROR.
In that case, it was indeed redundant, it was example just in case there was a string like:
```ts
[ERROR]: @Class#Method
Couldn't eat the fruit, apple was rotten
// long error stack
```

**Comment by Meowhal** [comment]
Oh, I am sorry that I took it wrong. I just wrote what came to mind. 
In some places I have already written some error message so that will be used instead of Error:.

By the way, the actual display format of the log is as follows, with the log category appended at the top.
`[20:40:30.123] [LEVEL] category - message `

I decided to use this category separately on a file-by-file basis.
If the category name is the class name, only the method name may be sufficient.
```ts
const logger = getLogger("ClassName");
logger.error(`@Method\n${e.message}\n${e.stack}`);
```
```log
[20:40:30.123] [ERROR] ClassName - @Method 
Couldn't eat the fruit, apple was rotten
// long error stack
```
Anyway I like the style where the first line contains only meta information.
